tests/AssertTest.php

```diff
@@ -307,6 +319,18 @@ public function getTests()
             array('endsWith', array('😄😑☹️', '☹️'), true),
             array('endsWith', array('😄😑☹️', '😑'), false),
             array('endsWith', array('', '😑'), false),
+            array('notEndsWith', array('abcd', 'cd'), false),
+            array('notEndsWith', array('abcd', 'bc'), true),
+            array('notEndsWith', array('', 'bc'), true),
+            array('notEndsWith', array('äþçð', 'çð'), false),
+            array('notEndsWith', array('äþçð', 'þç'), true),
+            array('notEndsWith', array('', 'þç'), true),
+            array('notEndsWith', array('あいうえ', 'うえ'), false),
+            array('notEndsWith', array('あいうえ', 'いう'), true),
+            array('notEndsWith', array('', 'いう'), true),
+            array('notEndsWith', array('😄😑☹️', '☹️'), false),
+            array('notEndsWith', array('😄😑☹️', '😑'), true),
+            array('notEndsWith', array('', '😑'), true),
```

Comment:
I'm wondering if this is 'correct'.
How do other assertion frameworks etc deal with an empty string in cases like this.?

Reply:
I believe it is correct to mirror the `xxWith()` functions since that's what would happen in the first place if you were to use `!`. If you look at [eclipse collections](https://github.com/eclipse/eclipse-collections/blob/master/unit-tests/src/test/java/org/eclipse/collections/impl/block/factory/StringPredicates2Test.java), that's also how they've implemented it.

Reply:
I think too because we have “stringNotEmpty” which handles such checks ?